### PR TITLE
core: Fix various syncing issue.

### PR DESCRIPTION
**Pull request by jimmyhu-dexon**
- Fix `MemBackedBlockDB.Update`. Should return an error if the block does not exist.
- Call `App.BlockConfirmed` when receiving finalized block.
- Tolerance lattice retry error.
- Append config to `Lattice` in sync mode.

### CODE REVIEW

File: core/consensus.go [hunk 1]
```diff
@@ -1082,6 +1085,8 @@ func (con *Consensus) processFinalizedBlock(block *types.Block) (err error) {
 				err = nil
 			}
 			con.lattice.ProcessFinalizedBlock(b)
+			con.logger.Debug("Calling Application.BlockConfirmed", "block", b)
+			con.app.BlockConfirmed(*b.Clone())
```
Comment (missionliao) [inline core/consensus.go]: What's the purpose to call `BlockConfirmed` when process finalized blocks?

File: core/lattice.go [hunk 2]
```diff
@@ -146,7 +146,6 @@ func (l *Lattice) SanityCheck(b *types.Block) (err error) {
 			if _, ok := err.(*ErrAckingBlockNotExists); ok {
 				err = ErrRetrySanityCheckLater
 			}
-			l.logger.Error("Sanity Check failed", "error", err)
```
Comment (missionliao) [inline core/lattice.go]: What's the purpose to remove this log?

File: core/consensus.go (hunk 1 above)
Comment (jimmyhu-dexon) [reply core/consensus.go]: `BlockConfirmed` is required to be called for all blocks including finalized blocks or Application won't know suck blocks.

File: core/lattice.go (hunk 2 above)
Comment (jimmyhu-dexon) [reply core/lattice.go]: The error is already reported outside. This will cause duplicated error messages.

File: core/consensus.go (hunk 1 above)
Comment (missionliao) [reply core/consensus.go]: If we don't call `BlockDelivered` for finalized blocks, is this call to `BlockConfirmed` still required?

**Pull request merged**